src/base/bittorrent/session.cpp

```diff
@@ -3585,6 +3592,25 @@ void Session::setMaxConcurrentHTTPAnnounces(const int value)
     configureDeferred();
 }
 
+bool Session::isReannounceWhenAddressChanged() const
+{
+    return m_isReannounceWhenAddressChanged;
+}
+
+void Session::setReannounceWhenAddressChanged(const bool enabled)
+{
+    if (enabled == m_isReannounceWhenAddressChanged)
+        return;
+
+    m_isReannounceWhenAddressChanged = enabled;
+}
+
+void Session::reannounceToAllTrackers()
+{
+    for (const lt::torrent_handle& torrent : m_nativeSession->get_torrents())
```

Comment:
In some places, torrent is checked for validiity i.e: https://github.com/qbittorrent/qBittorrent/blob/a51742b47c4eb81d81a7ccd1aab90832209f14b9/src/base/bittorrent/session.cpp#L3030-L3038

But then there is: https://github.com/qbittorrent/qBittorrent/blob/a51742b47c4eb81d81a7ccd1aab90832209f14b9/src/base/bittorrent/session.cpp#L4635-L4636

Dunno if it's intentional or an omission.

P.S. `&` should be attached to the variable.

Reply:
>Dunno if it's intentional or an omission.

I think the check isn't necessary here.

>P.S. & should be attached to the variable.

@zhuangzi926 
thalieht means it should be `const lt::torrent_handle &torrent`

Reply:
@zhuangzi926 
Seems you can also replace the code [here](https://github.com/qbittorrent/qBittorrent/blob/a51742b47c4eb81d81a7ccd1aab90832209f14b9/src/base/bittorrent/session.cpp#L4635-L4636) with your new class function.

Reply:
>if (!handle.is_valid()) continue; 

It is redundant there.

Reply:
Removed validness, and fix `&`.